fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItemSettings.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.item.v1;
+
+import net.minecraft.item.FoodComponent;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.util.Rarity;
+
+/**
+ * Fabric's version of Item.Settings. Adds additional methods and hooks
+ * not found in the original class.
+ *
+ * <p>To use it, simply replace {@code new Item.Settings()} with
+ * {@code new FabricItemSettings()}.
+ */
+public class FabricItemSettings extends Item.Settings {
+	private EquipmentSlotProvider equipmentSlotProvider;
+
+	/**
+	 * Sets the equipment slot provider of the item.
+	 *
+	 * @param equipmentSlotProvider the equipment slot provider
+	 * @return this builder
+	 */
+	public FabricItemSettings equipmentSlot(EquipmentSlotProvider equipmentSlotProvider) {
+		this.equipmentSlotProvider = equipmentSlotProvider;
+		return this;
+	}
+
+	/**
+	 * @deprecated Internal method.
+	 */
+	@Deprecated
+	public EquipmentSlotProvider getEquipmentSlotProvider() {
```

Comment:
I would do this the same was blocks with the ExtraData and BlockInternals:

see https://github.com/FabricMC/fabric/blob/deaf48319c6665f2492239d0fd909f893e63ca11/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/impl/object/builder/FabricBlockInternals.java#L57

Reply:
Done. I used a `WeakHashMap` instead of a mixin to `Item.Settings` as it's a lot shorter and doesn't need another mixin either.